src/pywifes/extract_spec.py

```diff
@@ -0,0 +1,516 @@
+    hdulist.close()
+
+
+def plot_apertures(data_cube, source_apertures, sky_aps=None, border_width=0):
+    """
+    Plot apertures on a collapsed data cube image.
+
+    Parameters
+    ----------
+    data_cube : numpy.ndarray
+        The 3D data cube containing the image data.
+
+    source_apertures : list of photutils.aperture objects
+        List of source apertures to plot on the image.
+
+    sky_aps : list of photutils.aperture objects, optional
+        List of sky apertures to plot on the image. Default is None.
+
+    border_width : int, optional
+        Width of the border to exclude from the image when calculating the image contrast.
+        Default is 0.
+
+    Returns
+    -------
+    None
+        This function does not return any value. It plots the apertures on the image.
+
+    """
+
+    # Collapse cube in the waevelenght dimesion for obtaing a median image
+    collapsed_cube = collapse_cube(data_cube)
+
+    # Improbe ccontrats on the image usefull for see faint sources
+    vmin, vmax = np.percentile(
+        collapsed_cube[border_width:-border_width, border_width:-border_width], (5, 95)
+    )
+    # Show collapsed image
+    plt.imshow(collapsed_cube, vmin=vmin, vmax=vmax)
+
+    # Color of the overlay area
+    cmap = mcolors.ListedColormap(["white"])
+    alpha = 0.5
+
+    for index, source_ap in enumerate(source_apertures):
+        ap_index = index + 1
+        # Plot a overlaped transparent area
+        mask = source_ap.to_mask(method="center").to_image(np.shape(collapsed_cube))
+        mask[mask == 0] = np.nan
+        plt.imshow(mask, alpha=alpha, cmap=cmap)
+        # Plot theoretical aperture contourngit
+        source_ap.plot(color="white", lw=0.8, ls="--")
+        # Plot the aperture number
+        plt.text(
+            source_ap.positions[0],
+            source_ap.positions[1],
+            ap_index,
+            color="white",
+            ha="center",
+            va="center",
+            fontsize=12,
+            path_effects=[withStroke(linewidth=2, foreground="black")],
+        )
+
+    if sky_aps is not None:
+        for sky_ap in sky_aps:
+            # Plot a overlaped transparent area
+            mask = sky_ap.to_mask(method="center").to_image(np.shape(collapsed_cube))
+            mask[mask == 0] = np.nan
+            plt.imshow(mask, alpha=alpha, cmap=cmap)
+            # Plot theoretical aperture contourn
+            sky_ap.plot(color="white", lw=0.8, ls="--")
+
+
+def read_cube_data(cube_path):
+    cube_data = {
+        "sci": None,
+        "sci_hdr": None,
+        "var": None,
+        "var_hdr": None,
+        "wcs": None,
+        "binning_x": None,
+        "binning_y": None,
+    }
+
+    if cube_path is not None:
+        sci, sci_hdr = fits.getdata(cube_path, 0, header=True)
+        var, var_hdr = fits.getdata(cube_path, 1, header=True)
+        wcs = WCS(sci_hdr).celestial
+        binning_x, binning_y = np.int_(fits.getheader(cube_path, 0)["CCDSUM"].split())
```

Comment:
`fits.getheader(cube_path, 0)` could just be `sci_hdr`.

Reply:
Done!